docs/api/utility-process.md

```diff
@@ -0,0 +1,128 @@
+
+* `modulePath` string - Absolute path to the script that should run as entrypoint in the child process.
+* `args` string[] (optional) - List of string arguments that will be available as `process.argv`
+  in the child process.
+* `options` Object (optional)
+  * `env` Object (optional) - Environment key-value pairs. Default is `process.env`.
+  * `execArgv` string[] (optional) - List of string arguments passed to the executable. Default is `process.execArgv`.
+  * `stdio` (string[] | string) (optional) - Child's stdout and stderr configuration. Default is `pipe`.
+    String value can be one of `pipe`, `ignore`, `inherit`, for more details on these values you can refer to
+    [stdio][] documentation from Node.js. Currently this option does not allow configuring
+    stdin and is always set to `ignore`. For example, the supported values will be processed as following:
+    * `pipe`: equivalent to ['ignore', 'pipe', 'pipe'] (the default)
+    * `ignore`: equivalent to 'ignore', 'ignore', 'ignore']
+    * `inherit`: equivalent to ['ignore', 'inherit', 'inherit']
+  * `serviceName` string (optional) - Name of the process that will appear in `name` property of
+    [`child-process-gone` event of `app`](app.md#event-child-process-gone).
+    Default is `node.mojom.NodeService`.
+  * `allowLoadingUnsignedLibraries` boolean (optional) _macOS_ - With this flag, the utility process will be
+    launched via the `Electron Helper (Plugin).app` helper executable on macOS, which can be
+    codesigned with `com.apple.security.cs.disable-library-validation` and
+    `com.apple.security.cs.allow-unsigned-executable-memory` entitlements. This will allow the utility process
+    to load unsigned libraries. Unless you specifically need this capability, it is best to leave this disabled.
+    Default is `false`.
+
+### Instance Methods
+
+#### `child.postMessage(message, [transfer])`
+
+* `message` any
+* `transfer` MessagePortMain[] (optional)
+
+Send a message to the child process, optionally transferring ownership of
+zero or more [`MessagePortMain`][] objects.
+
+For example:
+
+```js
+// Main process
+const { port1, port2 } = new MessageChannelMain()
+const child = new UtilityProcess(path.join(__dirname, 'test.js'))
+child.postMessage({ message: 'hello' }, [port1])
+
+// Child process
+process.parentPort.once('message', (e) => {
+  const [port] = e.ports
+  // ...
+})
+```
+
+#### `child.kill()`
+
+Returns `boolean`
+
+Terminates the process gracefully. On POSIX, it uses SIGTERM
+but will ensure to reap the process on exit. This function returns
+true if kill succeeds, and false otherwise.
+
+### Instance Properties
+
+#### `child.pid`
+
+A `Integer | undefined` representing the process identifier (PID) of the child process.
+If the child process fails to spawn due to errors, then the value is `undefined`.
+
+#### `child.stdout`
+
+A `NodeJS.ReadableStream | null | undefined` that represents the child process's stdout.
```

Comment:
is this possibly a bit of a footgun? i think the default here is `pipe`, and if the parent process does not properly read the pipe then the child process will stall when the stdout or stderr buffer fills up, which can happen due to logs in chromium or the child process itself.

Reply:
I feel like Node.js must have the same issue given their default is also `pipe` https://nodejs.org/dist/latest-v16.x/docs/api/child_process.html#optionsstdio. I flipped the default to `inherit` https://github.com/electron/electron/pull/34980/commits/6430fc8a68285f9499df08583308c87dd5b0ff0e